File: app/src/main/java/com.example.safehome/Publisher/Notification.java
package com.example.safehome.Publisher;

/**
 * Notification represents the abstract of a notification object.
 * All notifications implements this class.
 * @param <Data> represents the data about the event
 */
public interface Notification<Data> {

    /**
     * enables notifications
     */
    boolean enable();

    /**
     * disables notifications
     */
    boolean disable();
    /**

     * updates notification status
     */
    boolean update(Data data);

    /**
     * gets state from system
     */
    boolean getState();

    /**
     * sets state in system
     */
    void setState(boolean state);

    /**
     * sets type in system
     */
    void setType(EventType type);

    /**
     * gets type from system
     * @return {@linkplain EventType}
     */
    EventType getType();

    /**
     * gets data message received from an event
     * @return Data received from an event
     */
    Data getData();
}

File: app/src/main/kotlin/com.example.safehome/Publisher/Notification.kt
package com.example.safehome.Publisher

/**
 * Notification represents the abstract of a notification object.
 * All notifications implements this class.
 * @param <Data> represents the data about the event
 */

interface Notification {

    /**
     * enables notifications
     */
    fun enable(): Boolean

    /**
     * disables notifications
     */
    fun disable(): Boolean

    /**
     * updates notifications
     * @param data the data
     */
    fun update(data:Data?): Boolean

    /**
     * gets state from system
     */
    fun getState(): Boolean

    /**
     * sets state from system
     */
    fun setState(state:Boolean)

    /**
     * sets type in system
     */
    fun setType(type:EventType?)

    /**
     * gets type from system
     * @return EventType
     */
    fun getType(): EventType?

    /**
     * gets data message received from an event
     * @return Data received from an event
     */
    fun getData(): Data?

}